src/qtaudio/qtaudio.cc

```diff
@@ -205,29 +243,33 @@ void QtAudio::close_audio ()
     output_instance = nullptr;
 }
 
-void QtAudio::period_wait ()
+void QtAudioOutput::period_wait ()
 {
     pthread_mutex_lock (& mutex);
 
+#if QT_VERSION >= QT_VERSION_CHECK(6, 0, 0)
+    /* periodSize() is Qt 5 only, how to replace it? */
+#else
     while (output_instance->bytesFree () < output_instance->periodSize ())
```

Comment:
`period_wait` just needs to wait until `write_audio` will write *some* data (more than 0 bytes). So I think this can be as simple as:

```
while (output_instance->bytesFree () == 0)
```

It would be good to check that CPU usage doesn't spike to 100% (one core) or anything.

Reply:
The CPU usage is then still way too high. I assume because `pthread_cond_timedwait()` is used incorrectly. It expects an absolute time, not a relative one like in `static const timespec fifty_ms = {0, 50000000}`.

ALSA uses `clock_gettime()` instead. Is this approach fine to use in `period_wait()` and also `drain()`?

https://github.com/audacious-media-player/audacious-plugins/blob/ba579461cbec865c225e9ca1a32cf6f41b3fb5ad/src/alsa/alsa.cc#L466-L485

Reply:
Something like that should work, yes

Reply:
I have added a separate commit to fix the excessive CPU usage which also occurred with Qt 5.

A simpler fix would be to use `nanosleep()` but within pthread mutexes using `pthread_cond_timedwait()` should be better, right?

Reply:
Looks good. The advantage of pthread_timed_wait() is that it can return immediately if flush() is called. Just sleeping for 50ms could make seeking/stopping laggy.